error: could not find libv8_libplatform library

So, i've been reading through the other issues and posts, but nobody seems to provide a clear solution as to how to fix the above error;

Trying to install V8Js on a Debian distro, with PHP 7.0, using GN.

Did an apt-get upgrade, and installed all the deps.
Followed the instructions in README.Linux, but i end up with that compile error.

Please help

Same problem. Debian 8 x64, PHP 7.0.
```
sudo pecl install v8js
[sudo] password for tester: 
downloading v8js-1.4.0.tgz ...
Starting to download v8js-1.4.0.tgz (98,516 bytes)
......................done: 98,516 bytes
28 source files, building
running: phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012
Please provide the installation prefix of libv8 [autodetect] : 
building in /tmp/pear/temp/pear-build-rootEB4d1D/v8js-1.4.0
running: /tmp/pear/temp/v8js/configure --with-php-config=/usr/bin/php-config --with-v8js
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.5 (ok)
checking for gawk... gawk
checking for V8 Javascript Engine... yes, shared
checking for V8 files in default path... found in /usr
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for C standard version... c++11
checking how to allow c++11 narrowing...
checking for libv8_libplatform... configure: error: could not find libv8_libplatform library
ERROR: `/tmp/pear/temp/v8js/configure --with-php-config=/usr/bin/php-config --with-v8js' failed
```
Compiling myself from sources with fresh v8 in /opt/v8 ends the same. Looks like script is looking for libv8_libplatform library in wrong place. Or so I understood from looking script code.

Config logs for both cases distro and custom V8: [config.logs.zip](https://github.com/phpv8/v8js/files/1033474/config.logs.zip)

well, i was trying to install v8js on a Kali virtual machine, couldn't find a solution, so i did a 'dist-upgrade' and i ended up messing the whole machine.
now i got this:appstreamcli: symbol lookup error: /usr/lib/x86_64-linux-gnu/libxml2.so.2: undefined symbol: ucnv_close_57
reinstalling libxml doesn't seem to work... so i'm kinda stuck 

    On Saturday, May 27, 2017 1:31 PM, semoriil <notifications@github.com> wrote:
 

 Same problem. Debian 8 x64, PHP 7.0.sudo pecl install v8js
[sudo] password for tester: 
downloading v8js-1.4.0.tgz ...
Starting to download v8js-1.4.0.tgz (98,516 bytes)
......................done: 98,516 bytes
28 source files, building
running: phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012
Please provide the installation prefix of libv8 [autodetect] : 
building in /tmp/pear/temp/pear-build-rootEB4d1D/v8js-1.4.0
running: /tmp/pear/temp/v8js/configure --with-php-config=/usr/bin/php-config --with-v8js
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... re2c
checking for re2c version... 0.13.5 (ok)
checking for gawk... gawk
checking for V8 Javascript Engine... yes, shared
checking for V8 files in default path... found in /usr
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for C standard version... c++11
checking how to allow c++11 narrowing...
checking for libv8_libplatform... configure: error: could not find libv8_libplatform library
ERROR: `/tmp/pear/temp/v8js/configure --with-php-config=/usr/bin/php-config --with-v8js' failed
Compiling myself from sources with fresh v8 in /opt/v8 ends the same. Looks like script is looking for libv8_libplatform library in wrong place. Or so I understood from looking script code.Config logs for both cases distro and custom V8: config.logs.zip—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

Ok, i'm officially going crazy... been at it for 3 days now.

contents of config.log.

Gentoo v8js # cat config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by configure, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./configure --with-v8js=/opt/v8

## --------- ##
## Platform. ##
## --------- ##

hostname = Gentoo
uname -m = x86_64
uname -r = 4.9.16-gentoo
uname -s = Linux
uname -v = #1 SMP Sat May 27 05:21:17 EEST 2017

/usr/bin/uname -p = Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = unknown
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /tmp/v8/out.gn/x64.release
PATH: /opt/v8/
PATH: /tmp/depot_tools
PATH: /tmp/depot_tools
PATH: /usr/local/sbin
PATH: /usr/local/bin
PATH: /usr/sbin
PATH: /usr/bin
PATH: /sbin
PATH: /bin
PATH: /opt/bin
PATH: /usr/x86_64-pc-linux-gnu/gcc-bin/5.4.0
PATH: /tmp/depot_tools


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2468: checking for grep that handles long lines and -e
configure:2526: result: /bin/grep
configure:2531: checking for egrep
configure:2593: result: /bin/grep -E
configure:2598: checking for a sed that does not truncate output
configure:2652: result: /bin/sed
configure:2789: checking for cc
configure:2805: found /usr/bin/cc
configure:2816: result: cc
configure:2847: checking for C compiler version
configure:2856: cc --version >&5
gcc (Gentoo 5.4.0-r3 p1.3, pie-0.6.5) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:2867: $? = 0
configure:2856: cc -v >&5
Using built-in specs.
COLLECT_GCC=/usr/x86_64-pc-linux-gnu/gcc-bin/5.4.0/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-pc-linux-gnu/5.4.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /var/tmp/portage/sys-devel/gcc-5.4.0-r3/work/gcc-5.4.0/configure --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --prefix=/usr --bindir=/usr/x86_64-pc-linux-gnu/gcc-bin/5.4.0 --includedir=/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include --datadir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0 --mandir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/man --infodir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/info --with-gxx-include-dir=/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5 --with-python-dir=/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/python --enable-languages=c,c++,fortran --enable-obsolete --enable-secureplt --disable-werror --with-system-zlib --enable-nls --without-included-gettext --enable-checking=release --with-bugurl=https://bugs.gentoo.org/ --with-pkgversion='Gentoo 5.4.0-r3 p1.3, pie-0.6.5' --enable-libstdcxx-time --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-multilib --with-multilib-list=m32,m64 --disable-altivec --disable-fixed-point --enable-targets=all --disable-libgcj --enable-libgomp --disable-libmudflap --disable-libssp --disable-libcilkrts --disable-libmpx --enable-vtable-verify --enable-libvtv --enable-lto --without-isl --enable-libsanitizer
Thread model: posix
gcc version 5.4.0 (Gentoo 5.4.0-r3 p1.3, pie-0.6.5)
configure:2867: $? = 0
configure:2856: cc -V >&5
gcc: error: unrecognized command line option '-V'
gcc: fatal error: no input files
compilation terminated.
configure:2867: $? = 1
configure:2856: cc -qversion >&5
gcc: error: unrecognized command line option '-qversion'
gcc: fatal error: no input files
compilation terminated.
configure:2867: $? = 1
configure:2887: checking whether the C compiler works
configure:2909: cc    conftest.c  >&5
configure:2913: $? = 0
configure:2961: result: yes
configure:2964: checking for C compiler default output file name
configure:2966: result: a.out
configure:2972: checking for suffix of executables
configure:2979: cc -o conftest    conftest.c  >&5
configure:2983: $? = 0
configure:3005: result:
configure:3027: checking whether we are cross compiling
configure:3035: cc -o conftest    conftest.c  >&5
configure:3039: $? = 0
configure:3046: ./conftest
configure:3050: $? = 0
configure:3065: result: no
configure:3070: checking for suffix of object files
configure:3092: cc -c   conftest.c >&5
configure:3096: $? = 0
configure:3117: result: o
configure:3121: checking whether we are using the GNU C compiler
configure:3140: cc -c   conftest.c >&5
configure:3140: $? = 0
configure:3149: result: yes
configure:3158: checking whether cc accepts -g
configure:3178: cc -c -g  conftest.c >&5
configure:3178: $? = 0
configure:3219: result: yes
configure:3236: checking for cc option to accept ISO C89
configure:3299: cc  -c -g -O2  conftest.c >&5
configure:3299: $? = 0
configure:3312: result: none needed
configure:3338: checking how to run the C preprocessor
configure:3369: cc -E  conftest.c
configure:3369: $? = 0
configure:3383: cc -E  conftest.c
conftest.c:9:28: fatal error: ac_nonexistent.h: No such file or directory
compilation terminated.
configure:3383: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| #include <ac_nonexistent.h>
configure:3408: result: cc -E
configure:3428: cc -E  conftest.c
configure:3428: $? = 0
configure:3442: cc -E  conftest.c
conftest.c:9:28: fatal error: ac_nonexistent.h: No such file or directory
compilation terminated.
configure:3442: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| #include <ac_nonexistent.h>
configure:3473: checking for icc
configure:3482: result: no
configure:3496: checking for suncc
configure:3505: result: no
configure:3523: checking whether cc understands -c and -o together
configure:3551: cc -c conftest.c -o conftest2.o >&5
configure:3555: $? = 0
configure:3561: cc -c conftest.c -o conftest2.o >&5
configure:3565: $? = 0
configure:3620: result: yes
configure:3634: checking for system library directory
configure:3649: result: lib
configure:3656: checking if compiler supports -R
configure:3675: cc -o conftest -g -O2   conftest.c -R /usr/lib  >&5
gcc: error: unrecognized command line option '-R'
configure:3675: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
|
| int
| main ()
| {
|
|   ;
|   return 0;
| }
configure:3685: result: no
configure:3690: checking if compiler supports -Wl,-rpath,
configure:3709: cc -o conftest -g -O2   conftest.c -Wl,-rpath,/usr/lib  >&5
configure:3709: $? = 0
configure:3719: result: yes
configure:3764: checking build system type
configure:3778: result: x86_64-pc-linux-gnu
configure:3798: checking host system type
configure:3811: result: x86_64-pc-linux-gnu
configure:3831: checking target system type
configure:3844: result: x86_64-pc-linux-gnu
configure:3947: checking for PHP prefix
configure:3949: result: /usr/lib64/php7.0
configure:3951: checking for PHP includes
configure:3953: result: -I/usr/lib64/php7.0/include/php -I/usr/lib64/php7.0/include/php/main -I/usr/lib64/php7.0/include/php/TSRM -I/usr/lib64/php7.0/include/php/Zend -I/usr/lib64/php7.0/include/php/ext -I/usr/lib64/php7.0/include/php/ext/date/lib
configure:3955: checking for PHP extension directory
configure:3957: result: /usr/lib64/php7.0/lib/extensions/no-debug-non-zts-20151012
configure:3959: checking for PHP installed headers prefix
configure:3961: result: /usr/lib64/php7.0/include/php
configure:3964: checking if debug is enabled
configure:3990: result: no
configure:3993: checking if zts is enabled
configure:4019: result: no
configure:4090: checking for re2c
configure:4106: found /usr/bin/re2c
configure:4117: result: re2c
configure:4126: checking for re2c version
configure:4140: result: 0.16 (ok)
configure:4159: checking for gawk
configure:4176: found /usr/bin/gawk
configure:4187: result: gawk
configure:4230: checking for V8 Javascript Engine
configure:4270: result: yes, shared
configure:4505: checking for g++
configure:4521: found /usr/bin/g++
configure:4532: result: g++
configure:4559: checking for C++ compiler version
configure:4568: g++ --version >&5
g++ (Gentoo 5.4.0-r3 p1.3, pie-0.6.5) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:4579: $? = 0
configure:4568: g++ -v >&5
Using built-in specs.
COLLECT_GCC=/usr/x86_64-pc-linux-gnu/gcc-bin/5.4.0/g++
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-pc-linux-gnu/5.4.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /var/tmp/portage/sys-devel/gcc-5.4.0-r3/work/gcc-5.4.0/configure --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --prefix=/usr --bindir=/usr/x86_64-pc-linux-gnu/gcc-bin/5.4.0 --includedir=/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include --datadir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0 --mandir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/man --infodir=/usr/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/info --with-gxx-include-dir=/usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/include/g++-v5 --with-python-dir=/share/gcc-data/x86_64-pc-linux-gnu/5.4.0/python --enable-languages=c,c++,fortran --enable-obsolete --enable-secureplt --disable-werror --with-system-zlib --enable-nls --without-included-gettext --enable-checking=release --with-bugurl=https://bugs.gentoo.org/ --with-pkgversion='Gentoo 5.4.0-r3 p1.3, pie-0.6.5' --enable-libstdcxx-time --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-multilib --with-multilib-list=m32,m64 --disable-altivec --disable-fixed-point --enable-targets=all --disable-libgcj --enable-libgomp --disable-libmudflap --disable-libssp --disable-libcilkrts --disable-libmpx --enable-vtable-verify --enable-libvtv --enable-lto --without-isl --enable-libsanitizer
Thread model: posix
gcc version 5.4.0 (Gentoo 5.4.0-r3 p1.3, pie-0.6.5)
configure:4579: $? = 0
configure:4568: g++ -V >&5
g++: error: unrecognized command line option '-V'
g++: fatal error: no input files
compilation terminated.
configure:4579: $? = 1
configure:4568: g++ -qversion >&5
g++: error: unrecognized command line option '-qversion'
g++: fatal error: no input files
compilation terminated.
configure:4579: $? = 1
configure:4583: checking whether we are using the GNU C++ compiler
configure:4602: g++ -c   conftest.cpp >&5
configure:4602: $? = 0
configure:4611: result: yes
configure:4620: checking whether g++ accepts -g
configure:4640: g++ -c -g  conftest.cpp >&5
configure:4640: $? = 0
configure:4681: result: yes
configure:4709: checking how to run the C++ preprocessor
configure:4736: g++ -E  conftest.cpp
configure:4736: $? = 0
configure:4750: g++ -E  conftest.cpp
conftest.cpp:10:28: fatal error: ac_nonexistent.h: No such file or directory
compilation terminated.
configure:4750: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
| #include <ac_nonexistent.h>
configure:4775: result: g++ -E
configure:4795: g++ -E  conftest.cpp
configure:4795: $? = 0
configure:4809: g++ -E  conftest.cpp
conftest.cpp:10:28: fatal error: ac_nonexistent.h: No such file or directory
compilation terminated.
configure:4809: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
| #include <ac_nonexistent.h>
configure:4853: checking for C standard version
configure:4878: g++ -o conftest -g -O2 -std=c++11  -Wl,--rpath=/opt/v8/lib conftest.cpp -lstdc++  >&5
configure:4878: $? = 0
configure:4878: ./conftest
configure:4878: $? = 0
configure:4896: result: c++11
configure:4899: checking how to allow c++11 narrowing
configure:4928: g++ -o conftest -std=c++11   -Wl,--rpath=/opt/v8/lib conftest.cpp -lstdc++  >&5
conftest.cpp: In function 'int main()':
conftest.cpp:11:45: error: narrowing conversion of '-1' from 'int' to 'unsigned int' inside { } [-Wnarrowing]
         struct { unsigned int x; } foo = {-1};
                                             ^
configure:4928: $? = 1
configure: program exited with status 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
| int main() {
|         struct { unsigned int x; } foo = {-1};
|         (void) foo;
|         return 0;
|     }
configure:4947: g++ -o conftest -Wno-c++11-narrowing -std=c++11   -Wl,--rpath=/opt/v8/lib conftest.cpp -lstdc++  >&5
conftest.cpp: In function 'int main()':
conftest.cpp:11:49: error: narrowing conversion of '-1' from 'int' to 'unsigned int' inside { } [-Wnarrowing]
             struct { unsigned int x; } foo = {-1};
                                                 ^
At global scope:
cc1plus: warning: unrecognized command line option '-Wno-c++11-narrowing'
configure:4947: $? = 1
configure: program exited with status 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
| int main() {
|             struct { unsigned int x; } foo = {-1};
|             (void) foo;
|             return 0;
|         }
configure:4966: g++ -o conftest -Wno-narrowing -std=c++11   -Wl,--rpath=/opt/v8/lib conftest.cpp -lstdc++  >&5
configure:4966: $? = 0
configure:4966: ./conftest
configure:4966: $? = 0
configure:4998: result: -Wno-narrowing
configure:5017: checking for libv8_libplatform
configure:5042: g++ -o conftest -g -O2  -I/opt/v8/include -std=c++11  -Wl,--rpath=/opt/v8/lib -L/opt/v8/lib conftest.cpp -lstdc++   -lv8_libplatform -lv8 >&5
/tmp/ccCQ9Mj3.o: In function `main':
/tmp/v8js/conftest.cpp:21: undefined reference to `v8::platform::CreateDefaultPlatform(int, v8::platform::IdleTaskSupport)'
collect2: error: ld returned 1 exit status
configure:5042: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
|
|         namespace v8 {
|               namespace platform {
|                 enum class IdleTaskSupport { kDisabled, kEnabled };
|                 void* CreateDefaultPlatform(int thread_pool_size = 0, IdleTaskSupport idle_task_support = IdleTaskSupport::kDisabled);
|               }
|         }
|
| int
| main ()
| {
|  v8::platform::CreateDefaultPlatform();
|   ;
|   return 0;
| }
configure:5072: g++ -o conftest -g -O2  -I/opt/v8/include -std=c++11  -Wl,--rpath=/opt/v8/lib -L/opt/v8/lib conftest.cpp -lstdc++   -lv8_libplatform -lv8  -lv8_libplatform -lv8 >&5
/tmp/ccumecDc.o: In function `main':
/tmp/v8js/conftest.cpp:21: undefined reference to `v8::platform::CreateDefaultPlatform(int)'
collect2: error: ld returned 1 exit status
configure:5072: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
|
|         namespace v8 {
|               namespace platform {
|                 enum class IdleTaskSupport { kDisabled, kEnabled };
|                 void* CreateDefaultPlatform(int thread_pool_size = 0);
|               }
|         }
|
| int
| main ()
| {
|  v8::platform::CreateDefaultPlatform();
|   ;
|   return 0;
| }
configure:5102: g++ -o conftest -g -O2  -I/opt/v8/include -std=c++11  -Wl,--rpath=/opt/v8/lib -L/opt/v8/lib conftest.cpp -lstdc++   -lv8_libplatform -lv8  -lv8_libplatform -lv8 -lv8_libbase -lv8_libplatform -lv8 >&5
/tmp/ccbkdamn.o: In function `main':
/tmp/v8js/conftest.cpp:21: undefined reference to `v8::platform::CreateDefaultPlatform(int, v8::platform::IdleTaskSupport)'
collect2: error: ld returned 1 exit status
configure:5102: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
|
|         namespace v8 {
|               namespace platform {
|                 enum class IdleTaskSupport { kDisabled, kEnabled };
|                 void* CreateDefaultPlatform(int thread_pool_size = 0, IdleTaskSupport idle_task_support = IdleTaskSupport::kDisabled);
|               }
|         }
|
| int
| main ()
| {
|  v8::platform::CreateDefaultPlatform();
|   ;
|   return 0;
| }
configure:5132: g++ -o conftest -g -O2  -I/opt/v8/include -std=c++11  -Wl,--rpath=/opt/v8/lib -L/opt/v8/lib conftest.cpp -lstdc++   -lv8_libplatform -lv8  -lv8_libplatform -lv8 -lv8_libbase -lv8_libplatform -lv8 -lv8_libbase -lv8_libplatform -lv8 >&5
/tmp/cce6US0u.o: In function `main':
/tmp/v8js/conftest.cpp:21: undefined reference to `v8::platform::CreateDefaultPlatform(int)'
collect2: error: ld returned 1 exit status
configure:5132: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"
| /* end confdefs.h.  */
|
|         namespace v8 {
|               namespace platform {
|                 enum class IdleTaskSupport { kDisabled, kEnabled };
|                 void* CreateDefaultPlatform(int thread_pool_size = 0);
|               }
|         }
|
| int
| main ()
| {
|  v8::platform::CreateDefaultPlatform();
|   ;
|   return 0;
| }
configure:5141: error: could not find libv8_libplatform library

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_build=x86_64-pc-linux-gnu
ac_cv_c_compiler_gnu=yes
ac_cv_cxx_compiler_gnu=yes
ac_cv_env_CCC_set=
ac_cv_env_CCC_value=
ac_cv_env_CC_set=
ac_cv_env_CC_value=
ac_cv_env_CFLAGS_set=
ac_cv_env_CFLAGS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CPP_set=
ac_cv_env_CPP_value=
ac_cv_env_CXXCPP_set=
ac_cv_env_CXXCPP_value=
ac_cv_env_CXXFLAGS_set=
ac_cv_env_CXXFLAGS_value=
ac_cv_env_CXX_set=
ac_cv_env_CXX_value=
ac_cv_env_LDFLAGS_set=
ac_cv_env_LDFLAGS_value=
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=x86_64-pc-linux-gnu
ac_cv_objext=o
ac_cv_path_EGREP='/bin/grep -E'
ac_cv_path_GREP=/bin/grep
ac_cv_prog_AWK=gawk
ac_cv_prog_CPP='cc -E'
ac_cv_prog_CXXCPP='g++ -E'
ac_cv_prog_RE2C=re2c
ac_cv_prog_ac_ct_CC=cc
ac_cv_prog_ac_ct_CXX=g++
ac_cv_prog_cc_c89=
ac_cv_prog_cc_cc_c_o=yes
ac_cv_prog_cc_g=yes
ac_cv_prog_cxx_g=yes
ac_cv_target=x86_64-pc-linux-gnu
ac_cv_v8_cstd=c++11
ac_cv_v8_narrowing=-Wno-narrowing
lt_cv_path_SED=/bin/sed
php_cv_cc_dashr=no
php_cv_cc_rpath=yes
php_cv_re2c_version='0.16 (ok)'

## ----------------- ##
## Output variables. ##
## ----------------- ##

AR=''
AWK='gawk'
CC='cc'
CFLAGS='-g -O2'
CONFIGURE_COMMAND=' '\''./configure'\''  '\''--with-v8js=/opt/v8'\'''
CONFIGURE_OPTIONS=' '\''--with-v8js=/opt/v8'\'''
CPP='cc -E'
CPPFLAGS=' -I/opt/v8/include -std=c++11'
CXX='g++'
CXXCPP='g++ -E'
CXXFLAGS='-g -O2'
DEFS=''
DSYMUTIL=''
ECHO='echo'
ECHO_C=''
ECHO_N='-n'
ECHO_T=''
EGREP='/bin/grep -E'
EXEEXT=''
GREP='/bin/grep'
LDFLAGS=' -Wl,--rpath=/opt/v8/lib -L/opt/v8/lib'
LIBOBJS=''
LIBS='-lstdc++   -lv8_libplatform -lv8  -lv8_libplatform -lv8 -lv8_libbase -lv8_libplatform -lv8 -lv8_libbase -lv8_libplatform -lv8'
LIBTOOL=''
LN_S=''
LTLIBOBJS=''
NMEDIT=''
OBJEXT='o'
PACKAGE_BUGREPORT=''
PACKAGE_NAME=''
PACKAGE_STRING=''
PACKAGE_TARNAME=''
PACKAGE_URL=''
PACKAGE_VERSION=''
PATH_SEPARATOR=':'
RANLIB=''
RE2C='re2c'
SED='/bin/sed'
SHELL='/bin/sh'
SHLIB_DL_SUFFIX_NAME='so'
SHLIB_SUFFIX_NAME='so'
STRIP=''
ac_ct_CC='cc'
ac_ct_CXX='g++'
bindir='${exec_prefix}/bin'
build='x86_64-pc-linux-gnu'
build_alias=''
build_cpu='x86_64'
build_os='linux-gnu'
build_vendor='pc'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE}'
dvidir='${docdir}'
exec_prefix='NONE'
host='x86_64-pc-linux-gnu'
host_alias='x86_64-pc-linux-gnu'
host_cpu='x86_64'
host_os='linux-gnu'
host_vendor='pc'
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='/usr/lib64/php7.0'
program_transform_name='s,x,x,'
psdir='${docdir}'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
sysconfdir='${prefix}/etc'
target='x86_64-pc-linux-gnu'
target_alias=''
target_cpu='x86_64'
target_os='linux-gnu'
target_vendor='pc'

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME ""
#define PACKAGE_TARNAME ""
#define PACKAGE_VERSION ""
#define PACKAGE_STRING ""
#define PACKAGE_BUGREPORT ""
#define PACKAGE_URL ""
#define PHP_V8_EXEC_PATH "/opt/v8/lib/libv8.so"

configure: exit 1

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -37,7 +37,7 @@ Vagrant.configure("2") do |config|
     end
   }
 
-  %w{5.9.35}.each { |version|
+  %w{5.9.35 6.0.318}.each { |version|
     config.vm.define "v8-#{version}" do |i|
       i.vm.synced_folder ".", "/data/v8js"
 

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -75,7 +75,7 @@ Vagrant.configure("2") do |config|
 
       # Install to /opt/libv8-#{version}/
       sudo mkdir -p /opt/libv8-#{version}/{lib,include}
-      sudo cp out.gn/x64.release/lib*.so out.gn/x64.release/*_blob.bin /opt/libv8-#{version}/lib/
+      sudo cp out.gn/x64.release/lib*.so out.gn/x64.release/*_blob.bin out.gn/x64.release/icudtl.dat /opt/libv8-#{version}/lib/
       sudo cp -R include/* /opt/libv8-#{version}/include/
     SHELL
     end

--- a/config.m4
+++ b/config.m4
@@ -174,6 +174,23 @@ int main ()
   V8_SEARCH_BLOB([snapshot_blob.bin], [PHP_V8_SNAPSHOT_BLOB_PATH])
 
 
+  dnl
+  dnl  Check for v8::ArrayBuffer::Allocator::NewDefaultAllocator
+  dnl
+  AC_CACHE_CHECK([for v8::ArrayBuffer::Allocator::NewDefaultAllocator], ac_cv_has_default_allocator, [
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([
+      #include <v8.h>
+    ], [ v8::ArrayBuffer::Allocator::NewDefaultAllocator(); ])], [
+      ac_cv_has_default_allocator=yes
+    ], [ 
+      ac_cv_has_default_allocator=no
+    ])
+  ])
+  if test "x$ac_cv_has_default_allocator" = "xno"; then
+    AC_DEFINE([USE_INTERNAL_ALLOCATOR], [1],
+              [Define unless v8::ArrayBuffer::Allocator::NewDefaultAllocator is usable.])
+  fi
+
   AC_LANG_RESTORE
   LIBS=$old_LIBS
   LDFLAGS="$old_LDFLAGS"

--- a/config.m4
+++ b/config.m4
@@ -92,31 +92,22 @@ if test "$PHP_V8JS" != "no"; then
   AC_MSG_CHECKING([for libv8_libplatform])
   AC_DEFUN([V8_CHECK_LINK], [
     save_LIBS="$LIBS"
-	LIBS="$LIBS $1 -lv8_libplatform -lv8"
-	AC_LINK_IFELSE([AC_LANG_PROGRAM([
-	  namespace v8 {
-		namespace platform {
-		  enum class IdleTaskSupport { kDisabled, kEnabled };
-		  void* CreateDefaultPlatform($2);
-		}
-	  }
-	], [ v8::platform::CreateDefaultPlatform(); ])], [
-	  dnl libv8_libplatform.so found
-	  AC_MSG_RESULT(found)
-	  V8JS_SHARED_LIBADD="$1 -lv8_libplatform $V8JS_SHARED_LIBADD"
-      $3
-	], [ $4 ])
+    LIBS="$LIBS $1 -lv8_libplatform -lv8"
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([
+      #include <libplatform/libplatform.h>
+    ], [ v8::platform::CreateDefaultPlatform(); ])], [
+      dnl libv8_libplatform.so found
+      AC_MSG_RESULT(found)
+      V8JS_SHARED_LIBADD="$1 -lv8_libplatform $V8JS_SHARED_LIBADD"
+        $3
+    ], [ $4 ])
     LIBS="$save_LIBS"
   ])
 
-  V8_CHECK_LINK([], [int thread_pool_size = 0, IdleTaskSupport idle_task_support = IdleTaskSupport::kDisabled], [], [
-    V8_CHECK_LINK([], [int thread_pool_size = 0], [], [
-      V8_CHECK_LINK([-lv8_libbase], [int thread_pool_size = 0, IdleTaskSupport idle_task_support = IdleTaskSupport::kDisabled], [], [
-        V8_CHECK_LINK([-lv8_libbase], [int thread_pool_size = 0], [], [
-          AC_MSG_ERROR([could not find libv8_libplatform library])
-        ])
-	  ])
-	])
+  V8_CHECK_LINK([], [], [], [
+    V8_CHECK_LINK([-lv8_libbase], [], [], [
+      AC_MSG_ERROR([could not find libv8_libplatform library])
+    ])
   ])
 
 

--- a/v8js_class.cc
+++ b/v8js_class.cc
@@ -71,6 +71,7 @@ struct v8js_jsext {
 };
 /* }}} */
 
+#ifdef USE_INTERNAL_ALLOCATOR
 class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
 public:
 	virtual void* Allocate(size_t length) {
@@ -80,6 +81,7 @@ public:
 	virtual void* AllocateUninitialized(size_t length) { return malloc(length); }
 	virtual void Free(void* data, size_t) { free(data); }
 };
+#endif  /** USE_INTERNAL_ALLOCATOR */
 
 
 static void v8js_free_storage(zend_object *object) /* {{{ */
@@ -201,6 +203,10 @@ static void v8js_free_storage(zend_object *object) /* {{{ */
 	c->modules_base.~vector();
 
 	zval_ptr_dtor(&c->zval_snapshot_blob);
+
+#ifndef USE_INTERNAL_ALLOCATOR
+	delete c->create_params.array_buffer_allocator;
+#endif
 }
 /* }}} */
 
@@ -353,8 +359,12 @@ static PHP_METHOD(V8Js, __construct)
 
 	new (&c->create_params) v8::Isolate::CreateParams();
 
+#ifdef USE_INTERNAL_ALLOCATOR
 	static ArrayBufferAllocator array_buffer_allocator;
 	c->create_params.array_buffer_allocator = &array_buffer_allocator;
+#else
+	c->create_params.array_buffer_allocator = v8::ArrayBuffer::Allocator::NewDefaultAllocator();
+#endif
 
 	new (&c->snapshot_blob) v8::StartupData();
 	if (snapshot_blob) {

